# no-collapsible-if

ðŸ’¼ This rule is enabled in the âœ… `recommended` config.

<!-- end auto-generated rule header -->

Merging collapsible if statements increases the code's readability.

## Noncompliant Code Example

```javascript
if (x != undefined) {
  if (y === 2) {
    // ...
  }
}
```

## Compliant Solution

```javascript
if (x != undefined && y === 2) {
  // ...
}
```
